compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -913,6 +915,46 @@ fn llvm_asm_scalar_type<'ll>(cx: &CodegenCx<'ll, '_>, scalar: Scalar) -> &'ll Ty
     }
 }
 
+fn function_target_features<'ll>(builder: &Builder<'_, 'll, '_>) -> impl Iterator<Item = &'ll str> {
```

Comment:
You can use `self.tcx.asm_target_features(instance.def_id()).contains(&feature)` instead and avoid all this logic. It's used earlier in this file to convert unsupported registers to clobbers.